yellowbrick/model_selection/dropping_curve.py

```diff
@@ -0,0 +1,379 @@
+    train_scores_mean_ : array, shape (n_ticks,)
+        Mean training data scores for each training split
+    train_scores_std_ : array, shape (n_ticks,)
+        Standard deviation of training data scores for each training split
+    valid_scores_ : array, shape (n_ticks, n_cv_folds)
+        Scores on validation set.
+    valid_scores_mean_ : array, shape (n_ticks,)
+        Mean scores for each validation split
+    valid_scores_std_ : array, shape (n_ticks,)
+        Standard deviation of  scores for each validation split
+    Examples
+    --------
+    >>> from yellowbrick.model_selection import DroppingCurve
+    >>> from sklearn.naive_bayes import GaussianNB
+    >>> model = DroppingCurve(GaussianNB())
+    >>> model.fit(X, y)
+    >>> model.show()
+    Notes
+    -----
+    This visualizer is based on sklearn.model_selection.validation_curve
+    """
+
+    def __init__(
+        self,
+        estimator,
+        feature_sizes=DEFAULT_FEATURE_SIZES,
+        groups=None,
+        ax=None,
+        logx=False,
+        cv=None,
+        scoring=None,
+        n_jobs=None,
+        pre_dispatch='all',
+        random_state=None,
+        **kwargs
+    ):
+
+        # Initialize the model visualizer
+        super(DroppingCurve, self).__init__(estimator, ax=ax, **kwargs)
+
+        # Validate the feature sizes
+        feature_sizes = np.asarray(feature_sizes)
+        if feature_sizes.ndim != 1:
+            raise YellowbrickValueError(
+                "must specify 1-D array of feature sizes, '{}' is not valid".format(
+                    repr(feature_sizes)
+                )
+            )
+
+        # Set the metric parameters to be used later
+        self.feature_sizes = feature_sizes
+        self.groups = groups
+        self.logx = logx
+        self.cv = cv
+        self.scoring = scoring
+        self.n_jobs = n_jobs
+        self.pre_dispatch = pre_dispatch
+        self.random_state = random_state
+
+    def fit(self, X, y=None):
+        """
+        Fits the feature dropping curve with the wrapped model to the specified data.
+        Draws training and cross-validation score curves and saves the scores to the
+        estimator.
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Input vector, where n_samples is the number of samples and
+            n_features is the number of features.
+        y : array-like, shape (n_samples) or (n_samples, n_features), optional
+            Target relative to X for classification or regression;
+            None for unsupervised learning.
+        """
+        # Get feature_sizes in whole numbers
+        n_features = X.shape[-1]
+        if np.issubdtype(self.feature_sizes.dtype, np.integer):
+            assert (self.feature_sizes >= 0).all(), 'Expected feature sizes in [0, n_features]'
+            assert (self.feature_sizes <= n_features).all(), 'Expected feature sizes in [0, n_features]'
```

Comment:
In general, I'd prefer that we did not use assertions to raise `AssertionError` types, we generally expect that if Yellowbrick is raising an exception it is a subclass of [`YellowbrickError`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/exceptions.py#L21) - here a `YellowbrickTypeError` or `YellowbrickValueError` is probably most appropriate.